### DISCUSSION TITLE
Trace where keymap is coming from

### DISCUSSION THREAD

**Discussion by rwijtvliet** [LazyVim/LazyVim#2952] [Q&A]
In the default lazyvim config, the key `s` in normal mode in telescope launches `flash` to select a line to go to. I'd like to know, where this mapping is defined, so I can use it to in another keymap.

When in telescope normal mode, I can use `:map s`, which gives me a popup message mentioning `.local/share/nvim/lazy/telescope.nvim/lua/telescope/mappings.lua:289`. But when I have a look at this file/line, I don't find the specification directly, but rather the function that was called to set the keymap:
```lua
  vim.keymap.set(mode, key_bind, function()
    local ret = key_func(prompt_bufnr)
    vim.api.nvim_exec_autocmds("User", { pattern = "TelescopeKeymap" })
    return ret
  end, vim.tbl_extend("force", opts, { buffer = prompt_bufnr, desc = get_desc_for_keyfunc(key_func, opts) }))
```

**Can I somehow trace this stack to find where the left- and righthandsides are set?**

Or, is there another way to find out this information?

---

PS: for now, I have found a way to retain the functionality and remap, like so:

```lua
--lua/plugins/telescope.lua
return {
  "nvim-telescope/telescope.nvim",
  opts = function(_, opts)
    local actions = require("telescope.actions")
    return vim.tbl_deep_extend("force", opts, {
      defaults = {
        mappings = {
          n = {
            ["s"] = false,
            ["j"] = opts.defaults.mappings.n.s,
          },
        },
      },
    })
  end,
}
```

**Comment by dpetka2001** [comment]
[Here](https://github.com/LazyVim/LazyVim/blob/97480dc5d2dbb717b45a351e0b04835f138a9094/lua/lazyvim/plugins/editor.lua#L281-L310) is the Flash configuration for Telescope.

**Comment by rwijtvliet** [comment]
Ah, awesome. 2 follow-up questions if I may:

* How did you find this? 

* Is there another way to remap this, than how I did it above? I mean, without copying over the entire definition of the `flash` variable from the snipped you linked to.

**Reply by dpetka2001** [reply]
I've just been using LazyVim for a long time and knew about this change. But if you search for `flash.nvim` in the repo, you will see that its spec is in `/lua/lazyvim/plugins/editor.lua` and if you go through that file you will see what I linked in my previous comment.

The correct way to remap it would be the way you did it, otherwise you would have to copy over the `flash` function definition as you said.

**Comment by rwijtvliet** [comment]
Alright, many thanks! I just constantly run into situations where I think that there must be an easier way to accomplish them. But then it turns out, there isn't ;)

**Reply by dpetka2001** [reply]
Or maybe there is and I'm just not aware of it. I'm just a simple user like you with very little experience in programming. I just try to learn like you :smile:
